packages/@react-spectrum/label/src/StaticField.tsx

```diff
@@ -0,0 +1,180 @@
+import {useDateFormatter, useListFormatter, useNumberFormatter} from '@react-aria/i18n';
+import {useStyleProps} from '@react-spectrum/utils';
+
+interface StaticFieldBaseProps extends DOMProps, StyleProps, Omit<SpectrumLabelableProps, 'necessityIndicator' | 'isRequired'>, AriaLabelingProps {}
+
+type NumberValue = number | RangeValue<number>;
+interface NumberProps<T extends NumberValue> {
+    value: T,
+    formatOptions?: Intl.NumberFormatOptions
+}
+
+type DateTime = Date | CalendarDate | CalendarDateTime | ZonedDateTime | Time;
+type RangeDateTime = RangeValue<DateTime>;
+type DateTimeValue = DateTime | RangeDateTime;
+interface DateProps<T extends DateTimeValue> {
+    value: T,
+    formatOptions?: Intl.DateTimeFormatOptions
+}
+
+interface StringProps<T extends string> {
+    value: T
+}
+
+interface StringListProps<T extends string[]> {
+    value: T,
+    formatOptions?: Intl.ListFormatOptions
+}
+
+type StaticFieldProps<T> =
+    T extends NumberValue ? NumberProps<T> :
+    T extends DateTimeValue ? DateProps<T> :
+    T extends string[] ? StringListProps<T> :
+    T extends string ? StringProps<T> :
+    never;
+
+type SpectrumStaticFieldProps<T> = StaticFieldProps<T> & StaticFieldBaseProps;
+
+function StaticField<T>(props: SpectrumStaticFieldProps<T>, ref: RefObject<HTMLElement>) {
+  let {
+    value, 
+    formatOptions,
+
+    ...otherProps
+  } = props;
+  let {styleProps} = useStyleProps(otherProps);
+
+  return (
+    <div
+      {...styleProps}
+      ref={ref as RefObject<HTMLDivElement>}>
+      <Label
+        elementType="span">
+        <span>
+          {Array.isArray(props.value) && 
+            <FormattedStringList value={value as string[]} formatOptions={formatOptions as Intl.ListFormatOptions} />}
+
+          {typeof value === 'object' && 'start' in (value as any) && typeof (value as RangeValue<NumberValue>).start === 'number' && 
+            <FormattedNumber value={value as NumberValue} formatOptions={formatOptions as Intl.NumberFormatOptions}  />}
+
+          {typeof value === 'object' && 'start' in (value as any) && typeof (value as RangeValue<DateTimeValue>).start !== 'number' && 
+            <FormattedDateRange value={value as DateTimeValue} formatOptions={formatOptions as Intl.DateTimeFormatOptions} />}
+
+          {typeof value === 'number' && 
+            <FormattedNumber value={value as NumberValue} formatOptions={formatOptions as Intl.NumberFormatOptions} />}
+
+          {typeof value === 'object' && 'calendar' in (value as any) && 
+            <FormattedDate value={value as DateTimeValue} formatOptions={formatOptions as Intl.DateTimeFormatOptions} />}
+
+          {value instanceof Date && 
+            <FormattedDate value={value as DateTimeValue} formatOptions={formatOptions as Intl.DateTimeFormatOptions} />}
+          
+          {value instanceof Time && 
+            <FormattedDate value={value as DateTimeValue} formatOptions={formatOptions as Intl.DateTimeFormatOptions} />}
```

Comment:
whats the difference between these three? They result in the same FormattedDate component usage right?

Reply:
hmmm i think it's possible to combine them given the right conditional statement, but the thing with CalendarDate, CalendarDateTime, ZonedTimeDate, and Time is that they don't have one unique property that they all shared. for now, what i can come up with is `(typeof value === 'object' && ('calendar' in (value as any) || 'hour' in (value as any)) || (value instanceof Date)`. as for Date i wasn't too sure what to do about that so i just left it as` (value instanceof Date)`